test/integration/fake_upstream.cc

```diff
@@ -972,10 +975,15 @@ FakeHttpConnection::waitForInexactRawData(const char* data, std::string* out,
     }
     return false;
   };
-  if (!time_system_.waitFor(lock_, absl::Condition(&reached), timeout)) {
-    return AssertionFailure() << "timed out waiting for raw data";
+  // Because the connection must be read disabled to not auto-consume the
+  // underlying data, waitFor hangs with no events to force the time system to
+  // continue. Break it up into smaller chunks.
```

Comment:
Is the idea the calling waitFor() in smaller chunks causes time to move forward somehow? Does the event loop get pumped somehow? I suspect this all works, but I don't quite understand it.